src/render.rs

```diff
@@ -345,6 +363,63 @@ impl<'a> Renderer<'a> {
         Ok(output.trim_right().to_string())
     }
 
+    fn render_macro(&mut self, call_node: Node) -> TeraResult<String> {
+        if let MacroCall {namespace, name: macro_name, params: call_params} = call_node {
+            // To find the real namespace we just look at the last one we got.
+            // If we have self, we take the last one otherwise we push it into the vec of namespaces
+            let real_namespace = match namespace.as_ref() {
+                "self" => {
+                    // TODO: handle error if we don't have a namespace
+                    self.macro_namespaces
+                        .last()
+                        .expect("Open an issue with a template sample please (mention `self namespace macro`)!")
+                        .to_string()
+                },
+                _ => {
+                    // TODO: String doesn't have Copy trait, can we avoid double cloning?
+                    self.macro_namespaces.push(namespace.clone());
+                    namespace.clone()
```

Comment:
You don't need to `clone` here at all. Just use `.as_str` to get a pointer to the `String`.